### PR TITLE
Align composer config platform with requirements

### LINKED ISSUE #80: Use target platform reqs instead of --ignore-platform-reqs
At the moment, there are two problems:
 - Installing recipe on a system running php 7.x, where we need to support 5.x, will install modules / third party dependencies that don't support 5.x. This can be fixed by setting the config 'platform' option to specify a target.
 - Attempting to install a recipe which may include module dependencies not currently installed on the host computer will block setp. This can be fixed with `--ignore-platform-reqs`.
 
 Since both of those solutions are mutually exclusive, we can only use one or the other.
 
 A better solution is a per-recipe configurable "platform" option which declares the ideal lowest common denominator for each recipe. This would include a target PHP version, as well as a set of provided platform extensions. This would allow us to simultaneously solve both problems.

## PRs
 - [ ] #119

### PR DESCRIPTION
**Pull request by dnsl48**
Implement composer config.platform to emulate lowest requirements of PHP and extensions.

Thus, if composer defines requirement for php>=5.6, composer.lock will be built for PHP 5.6 even if you run cow on the system with php 7.2

Fixes #80

### CODE REVIEW

File: src/Utility/Composer.php [hunk 1]
```diff
@@ -133,6 +151,109 @@ protected static function getUpdateConfig($directory, $repository)
...
+            }
+        }
+
+        return $config;
+    }
+
+    /**
+     * Read the allowed versions defined in the composer format and return them ordered
+     *
+     * @param String $versionDefinition version definition
```
Comment (robbieaverill) [inline src/Utility/Composer.php]:
```suggestion
     * @param string $versionDefinition version definition
```

File: src/Utility/Composer.php [hunk 2]
```diff
@@ -133,6 +151,109 @@ protected static function getUpdateConfig($directory, $repository)
...
+                    } else {
+                        return 0;
+                    }
+                }
+            });
+
+            return $versions;
+        } else {
+            return [];
+        }
```
Comment (robbieaverill) [inline src/Utility/Composer.php]: I'm pretty sure there are utility methods you can use to sort versions/constraints - have you had a look at https://github.com/composer/semver?

Comment (dnsl48) [reply src/Utility/Composer.php]: yes, I was thinking about that one.
Unfortunately, we cannot reuse much from there in regards to parsing versions (at least without making things more complicated than it needs to be), and adding this 3rd party dependency for only comparing the versions seems to be an overkill to me...
On the other hand, we have this bit isolated in a private function, so it cannot spread across the app. If we need the semver functionality later some day, we could add the dependency and refactor this function as well.

Comment (robbieaverill) [reply src/Utility/Composer.php]: Sweet!

File: src/Utility/Composer.php [hunk 3]
```diff
@@ -67,25 +86,27 @@ public static function createProject(
...
+                $runner->runCommand([
+                    'composer',
+                    'config',
+                    '--unset',
+                    $option,
+                    '--working-dir',
+                    $directory,
+                ]);
+            }
         }
```
Comment (robbieaverill) [inline src/Utility/Composer.php]: This split is technically an API break, but I don't think we mind too much since cow is mostly an internal tool.

**Pull request merged**